docs/paper/dev/api/plugin-configs.md

```diff
@@ -0,0 +1,149 @@
+
+## Creating a config.yml
+
+By placing a `config.yml` file inside of your plugin, you can specify the default values for certain settings. This will be located in the `resources` directory:
+```
+example-plugin
+└── src
+    └── main
+        ├── java
+        └── resources
+            ├── config.yml
+            └── plugin.yml
+```
+
+Then, when your plugin is initialised you must save this resource into the plugins data directory so that a user can edit the values. 
+Here is an example of how you would do this in your plugin's `onEnable`:
+
+```java
+public class TestPlugin extends JavaPlugin {
+    
+    @Override
+    public void onEnable() {
+        saveResource("config.yml", /* replace */ false);
+
+        // You can also use this for configuration files:
+        saveDefaultConfig();
+        // Where the default config.yml will be saved if it does not already exist
+
+        // getConfig()...
+    }
+    
+}
+```
+
+:::info[`replace` parameter]
+
+The boolean `replace` parameter specifies whether it should replace an existing file if one exists. 
+If set to true, the configuration will be overwritten on every call.
+
+:::
+
+## Getting and setting data
+
+The `FileConfiguration` of the plugin can be fetched with `JavaPlugin#getConfig` once it has been saved. This will allow
+data to be fetched and set with the respective `#get...(key)` and `set(key, value)`. By Default, most basic data types are supported
+by YAML. These can be fetched simply with `#getString` or `#getBoolean`. 
+
+However, some more complex Bukkit data types are also supported. A few of these include `ItemStack`, `Location` and `Vector`s. 
+Here is an example of loading a value from the config for teleporting a player:
+
+:::info[Saving Configs]
+
+Whenever setting data in configurations, you must call `FileConfiguration#save` for the changes to persist on disk
+
+:::
+
+```java
+public class TestPlugin extends JavaPlugin {
+    public void teleportPlayer(Player player) {
+        Location to = getConfig().getLocation("target_location");
+        player.teleport(to);
+    }
+}
+```
+
+This is possible as they implement `ConfigurationSerializable`. You can use this yourself, by implementing and registering a custom class.
+
+```java
+public class TeleportOptions implements ConfigurationSerializable {
+    private int chunkX;
+    private int chunkZ;
+    private String name;
+    
+    public TeleportOptions(int chunkX, int chunkZ, String name) {
+        // Set the values
+    }
+    
+    public Map<String, Object> serialize() {
+        Map<String, Object> data = new HashMap();
+
+        data.put("chunk-x", this.chunkX);
+        data.put("chunk-y", this.chunkZ);
+        data.put("name", this.name);
+        
+        return data;
+    }
+    
+    public static TeleportOptions deserialize(Map<String, Object> args) {
+        return new TeleportOptions((int) args.get("chunk-x"), (int) args.get("chunk-x"), (String) args.get("name"));
```

Comment:
This deserialize two times the chunk X?

Reply:
Reseeing that why the name is chunk-y for a chunk Z?

Reply:
Good Point, fixed